dashboard/public/pixelation/pixelation.html

```diff
@@ -90,8 +90,42 @@
     $('#below_viz_instructions').text(options.instructions).show();
   }
   pixelationInit();
-  pixel_data.value = options.data;
-  drawGraph();
-  if(options.format=='raw') unformatBits()
-  else formatBitDisplay();
+
+  function pixelationDisplay() {
+    pixel_data.value = options.data;
+    drawGraph();
+    formatBitDisplay();
+  }
+
+  // Initialize projects for save/load functionality if channel id is present.
+  if (appOptions.channel) {
+    window.apps.setupProjectsExternal();
+    var projectsHandler = {
+      setInitialLevelHtml: function (levelHtml) {},
+      getLevelHtml: function () {
+        return '';
+      },
+      setInitialLevelSource: function (levelSource) {
+        options.data = levelSource;
```

Comment:
There are a few separate issues we are discussing here:
1. options being renamed to pixelationOptions
2. options/pixelationOptions (or even appOptions) being passed by template param rather than by js global
3. whether to merge pixelation params into ViewOptions/appOptions
4. whether ViewOptions should be used at all to pass params to pixelation or projects in the first place

one proposal is to pass everything via appOptions and use namespaces like appOptions.project, appOptions.pixelation, appOptions.view to compartmentalize (and maybe rename appOptions.level to appOptions.blocklyLevel if that's more accurate).

Let's discuss in person tomorrow as it doesn't seem like we're going to reach a consensus over github pull request comments. We don't have to do everything in this PR but it would be nice to agree on a direction.

to address your other concerns, for diffing: 
here is a branch which contains the same diff minus the pixelationOptions rename so you can review it more easily: https://github.com/code-dot-org/code-dot-org/compare/pixelation-save-state-no-rename

For the TODO:
once we agree on an approach, I can make the same changes to the other two external level types this week to eliminate the duplicate options state.